pytrip/utils/dddplot.py

```diff
@@ -0,0 +1,169 @@
+        """
+        if os.path.isfile(fn) is False:
+            raise IOError("Could not find file " + fn)
+
+        self.path = fn
+        self.filename = os.path.basename(fn)
+
+        with open(fn) as f:
+            lines = f.readlines()
+
+        for i, line in enumerate(lines):
+            _l = line.strip().split(' ', 1)
+            if len(_l) == 2:
+                key, value = _l
+            if key == "!filetype":
+                self.filetype = value
+            if key == "!fileversion":
+                self.fileversion = value
+            if key == "!filedate":
+                self.filedate = value
+            if key == "!projectile":
+                self.projectile = value
+            if key == "!material":
+                self.material = value
+            if key == "!composition":
+                self.composition = value
+            if key == "!density":
+                self.density = float(value)
+            if key == "!energy":
+                self.energy = float(value)
+
+        self.data = np.loadtxt(fn, comments=("#", "!"))
+
+        rows, cols = self.data.shape
+
+        if rows < 1:
+            raise IOError("Invalid number of data points in DDD file: " + fn)
+        self.ndata = rows
+
+        if cols < 2:
+            raise IOError("Invalid number of columns in DDD file: " + fn)
+        self.ngauss = cols - 2  # number of gaussian fits donw
+
+    def savefig(self, type="png"):
+        """
+        """
+        import matplotlib
+        matplotlib.use('Agg')
+        import matplotlib.pyplot as plt
+
+        # add suffix
+        outfile = self.filename + "." + type
+
+        fig, (p1, p2) = plt.subplots(2, 1)
+
+        p1.set(ylabel="dE/dz [MeV/(g/cm**2)]", title="{:s} - {:.3f} MeV/u".format(self.projectile, self.energy))
+        # p1.set(xlabel="z [g/cm**2]", ylabel="dE/dz [MeV/(g/cm**2)]")
+        p1.grid()
+        p1.plot(self.data[:, 0], self.data[:, 1])
+
+        if self.ngauss > 0:
+            p2.set(xlabel="z [g/cm**2]", ylabel="FWHM [g/cm**2]")
+            # p2.set_ylim([0, 2])
+            p2.set_yscale('log')
+            p2.grid(which="both")
+            p2.tick_params(axis='both', which='minor', labelsize=6)
+            # p2.yaxis.set_label_position("right")
+            # p2.yaxis.tick_right()
+            from matplotlib.ticker import FormatStrFormatter
+            p2.yaxis.set_minor_formatter(FormatStrFormatter('%.1f'))
+            for n in range(self.ngauss):
+                p2.plot(self.data[:, 0], self.data[:, n + 2])
+
+        plt.savefig(outfile)
+        plt.close()
+
+    @classmethod
+    def gauss(x, fwhm):
+        """ For future use
+        """
+        mu = 0
+        s = fwhm / 2.35482
+        return (1 / (s * np.sqrt(2 * np.pi)) * np.exp(- (x - mu)**2 / (2 * s**2)))
+
+
+def main(args=sys.argv[1:]):
+    """
+    Hint:
+    A series of .png files can be coverted to an animated gif using
```

Comment:
Animated GIF can be also saved with help of `pillow` library (it is automatically installed together wiht matplotlib). Some examples:
 - https://note.nkmk.me/en/python-pillow-gif/
 - https://www.c-sharpcorner.com/article/create-animated-gif-using-python-matplotlib/

Reply:
would do this in a future update